### PR TITLE
Adds time dependence to InterpolationTargetReceiveVars

### PR DESCRIPTION
**Pull request by markscheel**
## Proposed changes

This is a step towards having the horizon finder work in multiple frames.

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `major new feature` if appropriate.

### CODE REVIEW

**Review by kidder — DISMISSED** [review]

File: tests/Unit/NumericalAlgorithms/Interpolation/Test_InterpolationTargetReceiveVars.cpp [hunk 1]
```diff
@@ -271,30 +281,69 @@ struct MockMetavariables {
...
       mock_interpolation_target<MockMetavariables, InterpolationTargetA>,
       mock_interpolator<MockMetavariables>>;
   enum class Phase { Initialization, Testing, Exit };
 };
 
-template <typename MockCallbackType, size_t NumberOfExpectedCleanUpActions,
+// For updating the FunctionsOfTime.
+struct MyFunctionOfTimeUpdater {
+  static void apply(gsl::not_null<typename domain::Tags::FunctionsOfTime::type*>
+                        functions_of_time) {
```
Comment (moxcodes) [inline tests/Unit/NumericalAlgorithms/Interpolation/Test_InterpolationTargetReceiveVars.cpp]: could be `noexcept`

File: tests/Unit/NumericalAlgorithms/Interpolation/Test_InterpolationTargetReceiveVars.cpp [hunk 2]
```diff
@@ -271,30 +281,69 @@ struct MockMetavariables {
...
   using component_list = tmpl::list<
       mock_interpolation_target<MockMetavariables, InterpolationTargetA>,
       mock_interpolator<MockMetavariables>>;
   enum class Phase { Initialization, Testing, Exit };
 };
 
-template <typename MockCallbackType, size_t NumberOfExpectedCleanUpActions,
+// For updating the FunctionsOfTime.
+struct MyFunctionOfTimeUpdater {
+  static void apply(gsl::not_null<typename domain::Tags::FunctionsOfTime::type*>
```
Comment (moxcodes) [inline tests/Unit/NumericalAlgorithms/Interpolation/Test_InterpolationTargetReceiveVars.cpp]: `const`

File: tests/Unit/NumericalAlgorithms/Interpolation/Test_InterpolationTargetReceiveVars.cpp [hunk 3]
```diff
@@ -271,30 +281,69 @@ struct MockMetavariables {
...
   const size_t num_points = 10;
   const TimeStepId first_temporal_id(true, 0, Time(slab, Rational(13, 15)));
   const TimeStepId second_temporal_id(true, 0, Time(slab, Rational(14, 15)));
 
+  std::deque<TimeStepId> current_temporal_ids{};
+  std::deque<TimeStepId> pending_temporal_ids{};
+  std::unique_ptr<ActionTesting::MockRuntimeSystem<metavars>> runner_ptr{};
+  if constexpr (IsTimeDependent::value) {
+    const double expiration_time = 13.5 / 15.0;
+    current_temporal_ids.insert(current_temporal_ids.end(), first_temporal_id);
```
Comment (moxcodes) [inline tests/Unit/NumericalAlgorithms/Interpolation/Test_InterpolationTargetReceiveVars.cpp]: can you `push_back` for a `deque`?

File: tests/Unit/NumericalAlgorithms/Interpolation/Test_InterpolationTargetReceiveVars.cpp [hunk 4]
```diff
@@ -318,8 +365,8 @@ void test_interpolation_target_receive_vars() noexcept {
       {std::unordered_map<temporal_id_type, std::unordered_set<size_t>>{},
        std::unordered_map<temporal_id_type, std::unordered_set<size_t>>{
            invalid_indices},
-       std::deque<temporal_id_type>{},
-       std::deque<temporal_id_type>{first_temporal_id, second_temporal_id},
+       std::deque<temporal_id_type>{pending_temporal_ids},
```
Comment (moxcodes) [inline tests/Unit/NumericalAlgorithms/Interpolation/Test_InterpolationTargetReceiveVars.cpp]: Are the explicit copy constructors needed? I would have thought you could just pass `pending_temporal_ids` (same for next line)

File: tests/Unit/NumericalAlgorithms/Interpolation/Test_InterpolationTargetReceiveVars.cpp [hunk 5]
```diff
@@ -492,6 +540,69 @@ void test_interpolation_target_receive_vars() noexcept {
               intrp::Tags::CompletedTemporalIds<temporal_id_type>>(runner, 0)
               .front() == first_temporal_id);
 
+    if constexpr (IsTimeDependent::value) {
+      // There should zero TemporalIds left, but one PendingTemporalId.
```
Comment (moxcodes) [inline tests/Unit/NumericalAlgorithms/Interpolation/Test_InterpolationTargetReceiveVars.cpp:544]: These comments make the test very nicely readable :+1:

**Review by moxcodes — DISMISSED** [review]
looks good, I only have a few nitpicks -- feel free to squash any changes immediately.

**Comment by markscheel** [comment]
Made changes and squashed.

**Review by moxcodes — APPROVED** [review]

**Comment by markscheel** [comment]
The failing test is a roundoff-random-number problem with BarycentricRational that is apparently independent of this PR.

**Review by nilsdeppe — APPROVED** [review]

**Pull request merged**